rules/windows/credential_access_symbolic_link_to_shadow_copy_created.toml

```
@@ -0,0 +1,114 @@
+- Review priviledges of the end user as this requires administrative access.
+- Verify ntds.dit file was successfully copied and the location.
+- Investigate for registry SYSTEM file copies made recently or saved via Reg.exe.
+- Investigate recent deletions of volume shadow copies.
+- Identify other files potentially copied from volume shadow copy paths directly.
+
+### False Positive Analysis
+
+- There should be very little false positive triggers with this rule.
+
+### Related Rules
+- NTDS or SAM Database File Copied - 3bc6deaa-fbd4-433a-ae21-3e892f95624f
+
+### Response and Remediation
+
+- Initiate the incident response process based on the outcome of the triage.
+- In case specific credentials were compromised:
+    - Reset the password for the accounts
+- Locate and remove static files copied from volume shadow copies.
+- Command-Line tool mklink should require administrative access by default unless in developer mode.
+
+## Config
+
+Ensure advanced audit policies for Windows are enabled, specifically
+Object Access policies [Event ID 4656](https://docs.microsoft.com/en-us/windows/security/threat-protection/auditing/event-4656) (Handle to an Object was Requested)
+
+```
+Computer Configuration >
+Policies >
+Windows Settings >
+Security Settings >
+Advanced Audit Policies Configuration >
+System Audit Policies >
+Object Access >
+Audit File System (Success,Failure)
+Audit Handle Manipulation (Success,Failure)
+```
+
+This event will only trigger if symbolic links are created from a new process spawning for cmd.exe or powershell.exe with the correct arguments.
+Direct access to a shell and calling symbolic link creation tools will not generate an event.
+"""
+references = [
+    "https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/mklink",
+    "https://2017.zeronights.org/wp-content/uploads/materials/ZN17_Kheirkhabarov_Hunting_for_Credentials_Dumping_in_Windows_Environment.pdf",
+    "https://blog.netwrix.com/2021/11/30/extracting-password-hashes-from-the-ntds-dit-file/",
+    "https://www.hackingarticles.in/credential-dumping-ntds-dit/",
+]
+risk_score = 47
+rule_id = "d117cbb4-7d56-41b4-b999-bdf8c25648a0"
+severity = "medium"
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Credential Access"]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+sequence with maxspan=30s
+
+/* Create Symbolic Link to Shadow Copies */
+[process where event.type in ("start","process_created")
+    and (process.pe.original_file_name in ("Cmd.Exe","PowerShell.EXE")
+    and process.args : ("*mklink*","*SymbolicLink*")
+    and process.args : ("*\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy*"))]
+
+/* Copy NTDS file */
+[process where event.type in ("start","process_created")
+    and process.pe.original_file_name in ("Cmd.Exe","PowerShell.EXE")
+    and process.args : ("*Copy-Item*","*copy*","*mv*")
+    and process.args : "*ntds.dit"]
```

The second part of the sequence (copy ntds.dit) is already covered by [this](https://github.com/elastic/detection-rules/blob/main/rules/windows/credential_access_copy_ntds_sam_volshadowcp_cmdline.toml) rule. In general we need to minimize the use of sequence only when they are really necessary (minimize perf impact on the correlation engine). The first part of the sequence (mklink to shadow copy) should be a rare behavior in prod and not noisy (in itself is enough to alert on it, which also matches on the rule name `Symbolic Link to Shadow Copy Created`: 


```suggestion
process where event.type in ("start","process_created") and 
 process.pe.original_file_name in ("Cmd.Exe","PowerShell.EXE") and    
 
 /* Create Symbolic Link to Shadow Copies */
 process.args : ("*mklink*", "*SymbolicLink*") and process.command_line : ("*HarddiskVolumeShadowCopy*")
```